sdk/identity/Azure.Identity/src/EnvironmentCredentialProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    public class EnvironmentCredentialProvider : TokenCredential
+    {
+        private static readonly string s_AzureClientId = Environment.GetEnvironmentVariable("AZURE_CLIENT_ID");
```

Comment:
Assuming that we're intending this class to be used by external consumers, it may be nice to have a way to override the names of these variables, or to inject something responsible for reading them instead of doing so directly.

Reply:
This will likely change.  I'm not sure if eventually this will still read directly from the environment, or possibly from a shared azure config store.  But I don't really want to support other names for these environment variables.  The variable names I'm using here are dictated by the guidelines.

If a user wanted to read from another environment variable they could create their own variant of this provider, or they could read the variables and create the ClientSecretCredential directly.